### PR TITLE
Allow chunked uploads even if your quota is not sufficient

### LINKED ISSUE #11485: Uploading files in shared folders is broker with NC 14.0.1
### Steps to reproduce
0. **Edit:** The recipient needs to have a quota of 0
1. Share any folder with any account and give permissions to write.
2. Try to upload any file with the account the folder is shared with.

### Expected behaviour
Progress bar should appear and uploading starts.

### Actual behaviour
Uploading files in any shared folder isn’t posible. After choosing a file an click “ok” - nothing happens.
Uploading file in “real” folders is working.

### Server configuration

**Operating system**: Ubuntu 18.04

**Web server:** Apache

**Database:** MySQL

**PHP version:** PHP-FPM 7.2

**Nextcloud version:** 14.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated

**Where did you install Nextcloud from:** Web-Updater

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "toothless.goip.de",
            "toothless.fritz.box"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/toothless.goip.de",
        "dbtype": "mysql",
        "version": "14.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "enable_previews": false,
        "skeletondirectory": "",
        "logtimezone": "Europe\/Berlin",
        "log_type": "file",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "htaccess.RewriteBase": "\/",
        "filesystem_check_changes": 1,
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": 587,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "tls",
        "knowledgebaseenabled": false,
        "maintenance": false,
        "theme": "",
        "loglevel": 1,
        "data-fingerprint": "daf767cfda940da49aead6029558d893",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
}
```
</details>

**Are you using external storage, if yes which one:** No

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox

**Operating system:** Windows 10

### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Error	remote	TypeError: Argument 2 passed to OCA\DAV\Connector\Sabre\Directory::__construct() must implement interface OCP\Files\FileInfo, boolean given, called in /var/www/nextcloud/apps/dav/lib/Upload/UploadHome.php on line 93

    /var/www/nextcloud/apps/dav/lib/Upload/UploadHome.php - line 93:
    OCA\DAV\Connector\Sabre\Directory->__construct(OC\Files\View {}, false)

    /var/www/nextcloud/apps/dav/lib/Upload/UploadHome.php - line 51:
    OCA\DAV\Upload\UploadHome->impl()

    /var/www/nextcloud/apps/dav/lib/Upload/UploadHome.php - line 61:
    OCA\DAV\Upload\UploadHome->getChild("web-file-up ... 8")

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php - line 105:
    OCA\DAV\Upload\UploadHome->childExists("web-file-up ... 8")

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAVACL/Plugin.php - line 878:
    Sabre\DAV\Tree->nodeExists("uploads/Tes ... 8")

    /var/www/nextcloud/apps/dav/lib/Connector/Sabre/DavAclPlugin.php - line 95:
    Sabre\DAVACL\Plugin->beforeMethod(Sabre\HTTP\R ... "}, Sabre\HTTP\Response {})

    OCA\DAV\Connector\Sabre\DavAclPlugin->beforeMethod(Sabre\HTTP\R ... "}, Sabre\HTTP\Response {})

    /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105:
    call_user_func_array([ OCA\DAV\Co ... "], [ Sabre\HTTP ... }])

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 466:
    Sabre\Event\EventEmitter->emit("beforeMethod", [ Sabre\HTTP ... }])

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254:
    Sabre\DAV\Server->invokeMethod(Sabre\HTTP\R ... "}, Sabre\HTTP\Response {})

    /var/www/nextcloud/apps/dav/lib/Server.php - line 293:
    Sabre\DAV\Server->exec()

    /var/www/nextcloud/apps/dav/appinfo/v2/remote.php - line 35:
    OCA\DAV\Server->exec()

    /var/www/nextcloud/remote.php - line 163:
    require_once("/var/www/ne ... p")
```
</details>

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #11485

This allows uploads to shared folders.

To test just follow the steps in #11485

### CODE REVIEW

**Comment by rullzer** [comment]
@lubo @neeral85 @loeffelpan please have a look as well

File: lib/private/Files/Storage/Wrapper/Quota.php [hunk 1]
```diff
@@ -83,7 +83,7 @@ protected function getSize($path, $storage = null) {
 	 * @return int
 	 */
 	public function free_space($path) {
-		if ($this->quota < 0 || strpos($path, 'cache') === 0) {
+		if ($this->quota < 0 || strpos($path, 'cache') === 0 || strpos($path, 'uploads') === 0) {
```
Comment (MorrisJobke) [inline lib/private/Files/Storage/Wrapper/Quota.php]: That means once you have a folder `uploads` and no quota left you still could upload? 🧐

Comment (rullzer) [reply lib/private/Files/Storage/Wrapper/Quota.php]: No. Because that would be `files/uploads`

Comment (MorrisJobke) [reply lib/private/Files/Storage/Wrapper/Quota.php]: Ah ... missed that part. My brain was faster than I could read. It autocompleted to `!== false`

**Review by icewind1991 — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**